ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -287,12 +305,143 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			}
 			r.Log.Info("shutdownAfterJobFinishes set to true, we will delete cluster",
 				"RayJob", rayJobInstance.Name, "clusterName", fmt.Sprintf("%s/%s", rayJobInstance.Namespace, rayJobInstance.Status.RayClusterName))
-			return r.deleteCluster(ctx, rayJobInstance)
+			_, err = r.deleteCluster(ctx, rayJobInstance)
+			if err != nil && !errors.IsNotFound(err) {
+				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
+			}
+			_, err = r.deleteK8sJob(ctx, rayJobInstance)
+			if err != nil && !errors.IsNotFound(err) {
+				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
+			}
 		}
 	}
 	return ctrl.Result{}, nil
 }
 
+// CreateK8sJob creates a Kubernetes Job for the Ray Job if it doesn't exist, otherwise return the existing one.
+func (r *RayJobReconciler) CreateK8sJob(ctx context.Context, rayJobInstance *rayv1alpha1.RayJob) (string, error) {
+	var submitter_template v1.PodTemplateSpec
+	// Check if SubmitterPodTemplate is not provided
+	if rayJobInstance.Spec.SubmitterPodTemplate == nil {
```

Comment:
I just wanted `SubmitterPodTemplate` to be optional so that it's less of a burden for users to upgrade their YAML files to the new version.  For example, the v1alpha1 sample YAML doesn't need to be modified at all.

Reply:
I'm not confident that this is the correct way to handle optional fields though, let me know if you know a better way.